run/helm/snap/templates/_helpers.tpl

```diff
@@ -0,0 +1,36 @@
+{{/* vim: set filetype=mustache: */}}
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+*/}}
+{{- define "fullname" -}}
+{{- $name := default .Chart.Name .Values.nameOverride -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
```

Comment:
Comment added for "template" function.

As for "hash" function I'm removing it, sorry. It is legacy of my efforts to use rolling-upgrade on config change (automatic restart of pods on changes in configmap). Unfortunately rolling-upgrades are not supported yet in DaemonSets, only Deployments.